packages/samples/headless-ssr/src/app/generic/hooks/search-parameters.ts

```diff
@@ -0,0 +1,111 @@
+'use client';
```

Comment:
Can add unit tests for these hooks and new controllers (using `@testing-library/react`) - maybe as a followup PR?

Reply:
Sure